packages/web-react/src/components/TextFieldBase/TextFieldBase.tsx

```diff
@@ -25,20 +25,27 @@ export const TextFieldBase = (props: SpiritTextFieldBaseProps) => {
   });
   useDeprecationMessage({
     method: 'custom',
-    trigger: !!(props?.message && !props?.validationState),
+    trigger: !!(props?.message && !validationState),
     componentName: props?.isMultiline ? 'TextArea' : 'TextField',
     customText:
       '`message` prop without `validationState` prop will be unsupported in the next version. Use `helperText` instead.',
   });
 
+  const renderValidationText = useValidationText({
```

Comment:
I find this interesting (https://blog.bitsrc.io/new-react-design-pattern-return-component-from-hooks-79215c3eac00), but not sure if this is an anti-pattern.

Reply:
It is the same as in FileUploader. And this is your field of expertise, so feel free to suggest better solution.